uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Page/UserMenu.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+use CRM_HRCore_CMSData_PathsFactory as PathsFactory;
+
+class CRM_HRCore_Page_UserMenu extends CRM_Core_Page {
+
+  private $contactData;
+  private $cmsPaths;
+
+  public function run() {
+    $this->getContactData();
+    $this->instantiateCmsPaths();
+
+    $this->assign('username', $this->contactData['display_name']);
+    $this->assign('image', $this->getUserImagePath());
+    $this->assign('editLink', $this->cmsPaths->getEditAccountPath());
+    $this->assign('logoutLink', $this->cmsPaths->getLogoutPath());
+
+    return parent::run();
+  }
+
+  /**
+   * Gets the currently logged in contact's data, including the
+   * her user id in the CMS
+   *
+   * @return array
+   */
+  private function getContactData() {
+    $rawContactData = civicrm_api('Contact', 'getsingle', array(
+      'version' => 3,
+      'return' => array('id', 'display_name', 'image_URL'),
+      'id' => CRM_Core_Session::getLoggedInContactID(),
+      'api.User.getsingle' => array('contact_id' => "\$value.contact_id")
+    ));
+
+    $this->contactData = $this->normalizeContactDataAPIResponse($rawContactData);
```

Comment:
this method is better to be used near to were it is defined, so I would rather define normalizeContactDataAPIResponse directly below getContactData to avoid jumping while reading the code.